org.eclipse.lsp4e/src/org/eclipse/lsp4e/format/FormatStrategy.java

```diff
@@ -0,0 +1,31 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   See git history
+ *******************************************************************************/
+package org.eclipse.lsp4e.format;
+
+/**
+ * Format strategy to be applied to a document.
+ *
+ */
+public enum FormatStrategy {
```

Comment:
If you split 3 different implementation from FormatRegionsProvider, then you can decide to return 1 of this implementation directly.

Reply:
Good point. But I am little unsure how to handle this public interface, which a foreign bundle has to implement:
```java
public interface IFormatOnSave {

	/**
	 * Returns the {@link FormatStrategy} for the given document.
	 * @param document
	 * @return FormatStrategy
	 */
	FormatStrategy getFormatStrategy(IDocument document);

}
```
What should be the retuern type of `getFormatStrategy` when the foreign bundle should not implement `IFormatRegionsProvider`?

Reply:
I think I get your point here.